docs/ovs_communication.adoc

```diff
@@ -45,6 +45,20 @@ TBD - Eric
 . ofctl.api.send_msg (packet) (OSKen)
 
 TBD - XiaoDong, we want to get to the next level of detail. Imagine we are going to start writing the code next week, let's give an example of the calls to make to configuration a port for vxlan. We want to understand the options/parameters needed so that we can check to make the network state message from Alcor Controller contains that information.
+A: I think you want to do the cpp call to create vxlan, in this case please refer to ovs api vxlan_create and vxlan_tnl_create (struct vport * vxlan_tnl_create(const struct vport_parms *)) locating in ovs/datapath/vport-vxlan.c on github ovs project. To find more about the parameters passing into the function call, here is one examples:
+Parms -> options parms -> options Assign to options.
+  Assign to dev.
+  Assign to vport.
+  Assign to a.
+  Assign to err.
+  Assign to conf.
```

Comment:
I don't think ACA can call into the OVS internal code directly. Please correct me if I am wrong.

Reply:
what do you mean ACA?
the flow should be:
bridge_reconfigure->bridge_add_ports->bridge_add_ports__->iface_create->iface_do_create->ofproto_port_add->port_add(ofproto_class)->dpif_port_add->dpif_netlink_port_add(dpif_class)->dpif_netlink_port_add__-
in the kernel:
>ovs_vport_cmd_new->new_vport->ovs_vport_add --> vxlan_tnl_create

Reply:
what do you mean ACA?
the flow should be:
bridge_reconfigure->bridge_add_ports->bridge_add_ports__->iface_create->iface_do_create->ofproto_port_add->port_add(ofproto_class)->dpif_port_add->dpif_netlink_port_add(dpif_class)->dpif_netlink_port_add__-
in the kernel:
>ovs_vport_cmd_new->new_vport->ovs_vport_add --> vxlan_tnl_create